### ISSUE TITLE
Handling multiple errors

### ISSUE THREAD

**Issue by ethanresnick** [json-api/json-api#608]
There are cases in which the current spec cannot be interpreted literally because it requires simultaneously responding two different status codes, which is obviously impossible. Consider the following request to a server that doesn't support client-generated ids:

```
POST /people
Content-Type: application/vnd.api+json

{
  "data": {
    "type": "photos",
    "id": 550e8400-e29b-41d4-a716-446655440000",
    "attributes": {}
  }
}
```

The spec simultaneously requires here that the server:
1. "**MUST** return 403 Forbidden in response to an unsupported request to create a resource with a client-generated ID." and
2. "**MUST** return 409 Conflict when processing a POST request in which the resource's type does not match the server's endpoint."

So what's the appropriate behavior here?

My inclination is that we should say here what we're already saying in the JSON-PATCH extension (and actually that we should move the JSON-PATCH copy into the base spec). Namely, the server should have the choice _either_ to stop processing the request and respond with the appropriate status code for whichever error it happened to encounter first, _or_ to process the request all the way through, pick a generic status code, and return an array of errors.

**Comment by dgeb** [comment]
> My inclination is that we should say here what we're already saying in the JSON-PATCH extension (and actually that we should move the JSON-PATCH copy into the base spec). Namely, the server should have the choice either to stop processing the request and respond with the appropriate status code for whichever error it happened to encounter first, or to process the request all the way through, pick a generic status code, and return an array of errors.

:+1:

**Comment by tkellen** [comment]
@ethanresnick Have the bandwidth to tackle a PR for this?

**Comment by ethanresnick** [comment]
@tkellen I'll try later tonight, but no guarantees. Like I said, all my family's in town, so I don't have much time, and I'm much more focused on trying to write up something for #614 (which will take a ton of copy, but it ultimately seems like the most important issue, since it determines what we can get wrong now without too many consequences).

**Comment by tkellen** [comment]
Resolved in #655.
